server/graphql/requisition/src/mutations/response_requisition/insert.rs

```diff
@@ -0,0 +1,341 @@
+use async_graphql::*;
+use graphql_core::{
+    simple_generic_errors::{OtherPartyNotACustomer, OtherPartyNotVisible},
+    standard_graphql_error::validate_auth,
+    standard_graphql_error::StandardGraphqlError,
+    ContextExt,
+};
+use graphql_types::types::RequisitionNode;
+use repository::Requisition;
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    requisition::response_requisition::{
+        InsertResponseRequisition as ServiceInput, InsertResponseRequisitionError as ServiceError,
+    },
+};
+
+#[derive(InputObject)]
+#[graphql(name = "InsertResponseRequisitionInput")]
+pub struct InsertInput {
+    pub id: String,
+    pub other_party_id: String,
+    pub max_months_of_stock: f64,
+    pub min_months_of_stock: f64,
+}
+
+#[derive(Interface)]
+#[graphql(name = "InsertResponseRequisitionErrorInterface")]
```

Comment:
I wonder do we rename graphql fields throughout the code? I find the renaming slightly confusing to trace types, but maybe that is just me?

Reply:
Yes, we use this for the Object name throughout the codebase.

Reply:
(Not requesting any change!)

>Yes, we use this for the Object name throughout the codebase.

You mean the frontend code base? I think Ferg's point is that when trying to trace backend implementation it's not always intuitive to search for `InsertErrorInterface` which is generically named within this code context, but in a code search things become harder to find (and when working you have to be constantly aware of directory structure of the file you're editing to apply context to the variable name - bit of a cognitive load when trying to jump around different files.

![image](https://github.com/user-attachments/assets/dea0ea54-75f7-45c7-9178-6ada0720621b)

Trade off though - once you are contextualised the short names are cozy and it favours copy pasting all this boiler plate.

That said not everywhere has the different names:

https://github.com/msupply-foundation/open-msupply/blob/7a31855c50d62f04c22ac8014bc10bf3abdef6e5/server/graphql/requisition/src/mutations/request_requisition/insert_program.rs#L17-L19

Reply:
Ah that's true. I usually go through by folder name then action to find them 🤔